packages/iris-grid/src/PartitionSelectorSearch.tsx

```diff
@@ -275,19 +281,29 @@ class PartitionSelectorSearch<T> extends Component<
   }
 
   updateFilter(): void {
-    const { dh, initialPageSize, table } = this.props;
+    const { initialPageSize, table } = this.props;
     const { text } = this.state;
     const filterText = text.trim();
     const filters = [];
     if (filterText.length > 0) {
       const column = table.columns[0];
-      const filter = column
-        .filter()
-        .invoke(
-          'matches',
-          dh.FilterValue.ofString(`(?s)(?i).*\\Q${filterText}\\E.*`)
+      if (
+        column.type === 'java.lang.String' ||
+        column.type === 'char' ||
+        column.type === 'java.lang.Boolean' ||
+        !Number.isNaN(Number(filterText))
+      ) {
```

Comment:
Use `TableUtils.isTextType` etc to check the column type...
Instead of having a check like this though, it would be better if we validated the input when it is changed; for example, the `input` field could change to `type="number"` with an appropriate `step` if it's a integer or a double type, and `handleTextChange` could filter it so that only numbers from the input are accepted. Then we shouldn't need a check here.